### PR TITLE
feat: Support publicPath ending with '/'

### PR DESCRIPTION
**Pull request by chen-jingjie**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export function dynamicBase(options?: Options): Plugin {
   let assetsDir = 'assets'
   let base = '/'
   let legacy = false
-  let baseOptions: TransformOptions = { assetsDir, base, legacy, publicPath, transformIndexHtml }
+  let baseOptions: TransformOptions = { assetsDir, base, legacy, publicPath: `(${publicPath} || '').replace(/\\/*$/, '')`, transformIndexHtml }
```
Comment (chenxch) [inline src/index.ts:17]: Why use string templates here? And use error, please check and test.

**Review by chenxch — APPROVED** [review]
Can you tell me why you don't remove it when it's passed in?

**Comment by chen-jingjie** [comment]
我的场景是：我们项目是微前端项目 baseurl 是在基座上。部分项目是 webpack编译的。baseurl 已经使用了 带“/” 的路径。因此我不太想去改动这个变量。
另外：带“/”和不带都能支持其实是更完善的。类似于 nodejs 的 path.join 或者是 浏览器的new url（path1，path2）。这些 api 在设计的时候都是会兼容“/”的
----------

该邮件从移动设备发送


--------------原始邮件--------------
发件人："Xc ***@***.***&gt;;
发送时间：2022年7月30日(星期六) 晚上11:37
收件人："chenxch/vite-plugin-dynamic-base" ***@***.***&gt;;
抄送："陈敬杰 ***@***.***&gt;;"Author ***@***.***&gt;;
主题：Re: [chenxch/vite-plugin-dynamic-base] feat: Support publicPath ending with '/' (PR #11)
-----------------------------------

 


@chenxch approved this pull request.批准了这个请求。

Can you tell me why you don't remove it when it's passed in?你能告诉我为什么你不删除它时，它的传递？


In 在src/index.ts:
&gt; @@ -14,7 +14,7 @@ export function dynamicBase(options?: Options): Plugin {    let assetsDir = 'assets'    let base = '/'    let legacy = false -  let baseOptions: TransformOptions = { assetsDir, base, legacy, publicPath, transformIndexHtml } +  let baseOptions: TransformOptions = { assetsDir, base, legacy, publicPath: `(${publicPath} || '').replace(/\\/*$/, '')`, transformIndexHtml } 
Why use string templates here? And use error, please check and test.为什么要在这里使用字符串模板？和使用错误，请检查和测试。

—&amp;mdash;

**Comment by chenxch** [comment]
Ok, I see. There is a problem in your pr to solve it.

File: src/index.ts (hunk 1 above)
Comment (chenxch) [reply src/index.ts:17]: You are right.

**Comment by chenxch** [comment]
There are still some problems in the static resources part, such as `` `src` under the about page in the demo.

**Comment by chenxch** [comment]
Among them, `transformHtml` and `transformLegacyHtml` can be optimized.

**Comment by chen-jingjie** [comment]
> Among them, `transformHtml` and `transformLegacyHtml` can be optimized.

> There are still some problems in the static resources part, such as `` `src` under the about page in the demo.

These issues don't seem to be related to this PR

**Comment by chenxch** [comment]
> These issues don't seem to be related to this PR

`img` how can this be irrelevant?

**Comment by chen-jingjie** [comment]
> > These issues don't seem to be related to this PR
> 
> `img` how can this be irrelevant?

I mean this is how he used to be

**Comment by chen-jingjie** [comment]
我是不是没有说清楚？demo 里面的 img 无法访问不是由我的 PR 引起的。他原本就不可访问。

**Comment by chenxch** [comment]
为什么会本来就不能访问呢0 0

我说的是这个问题
你的场景你可以使用以下配置完全可以解决
```js
dynamicBase({ transformIndexHtml: true, publicPath: 'window.__dynamic_base__.replace(/\\/*$/, "")' }),
```

**Pull request closed without merging**